client/verta/tests/test_model_registry/test_model_version.py

```diff
@@ -423,6 +377,64 @@ def test_patch(self, registered_model):
 
         assert version.get_attributes() == ATTRIBUTES
 
+
+class TestFind:
+    def test_list_from_client(self, client, created_entities):
+        """
+        At some point, backend API was unexpectedly changed to require model ID
+        in /model_versions/find, which broke client.registered_model_versions.
+
+        """
+        registered_model = client.create_registered_model()
+        created_entities.append(registered_model)
+
+        len(client.registered_model_versions)
+
+    def test_find(self, client, created_entities):
+        name = "registered_model_test"
+        registered_model = client.set_registered_model()
+        created_entities.append(registered_model)
+        model_version = registered_model.get_or_create_version(name=name)
+
+        find_result = registered_model.versions.find(["version == '{}'".format(name)])
+        assert len(find_result) == 1
+        for item in find_result:
+            assert item._msg == model_version._msg
+
+        tag_name = name + "_tag"
+        versions = {name + "1": registered_model.get_or_create_version(name + "1"),
+                    name + "2": registered_model.get_or_create_version(name + "2")}
+        versions[name + "1"].add_label(tag_name)
+        versions[name + "2"].add_label(tag_name)
+        versions[name + "2"].add_label("label2")
+
+        for version in versions:
+            versions[version] = registered_model.get_version(version)
+
+        find_result = registered_model.versions.find(["labels == \"{}\"".format(tag_name)])
```

Comment:
The existence of this method rings the "dangerously clever" alarm bells in my head. That gut intuition isn't necessarily correct but I was surprised. The wisdom of this `find` method is beyond the scope of this review though and it has clear utility.